lib/resque/plugins/lock.rb

```diff
@@ -53,13 +53,14 @@ def lock_timeout
       # passed the same arguments as `perform`, that is, your job's
       # payload.
       def lock(*args)
-        "lock:#{name}-#{encode(*args)}"
+        "lock:#{name}-#{args.to_s}"
       end
 
       # See the documentation for SETNX http://redis.io/commands/setnx for an
       # explanation of this deadlock free locking pattern
       def before_enqueue_lock(*args)
-        key = lock(*args)
+        normalized_args = Resque.decode(Resque.encode(args))
```

Comment:
Don't we want to do this normalization in the around_perform_lock as well when deleting the lock ?